### PR TITLE
Chores/contacts view clean up

### PR DESCRIPTION
**Pull request by nuritnt**
- [x] Story under test
- [x] Mobile works
- [x] Seeds created
- [x] Translated

Review by Team
- [ ] Reviewed by vok

Review by Coach
- [ ] Reviewed by sta

### CODE REVIEW

File: app/controllers/clients_controller.rb [hunk 1]
```diff
@@ -49,11 +51,13 @@ def set_client
...
-      :permit, :gender, :street, :zip, :city, :phone, :email, :goals, :education, :hobbies,
+    params.require(:client).permit(:date_of_birth, :nationality,
+      :permit, :gender, :goals, :education, :hobbies,
       :interests, :state, :comments, :competent_authority, :involved_authority, :user_id,
       language_skills_attributes: language_skills_attributes,
       relatives_attributes: relatives_attributes,
-      schedules_attributes: schedules_attributes)
+      schedules_attributes: schedules_attributes,
+      contact_attributes: contact_attributes
+      )
```
Comment (casaper) [inline app/controllers/clients_controller.rb]: in the parameter returning methods, do:

```ruby
def contact_attributes
  { contact_attributes: [:bogus, :example, :more, :attr] }
end
```

instead of:
```ruby
def contact_attributes
  [:bogus, :example, :more, :attr]
end
```

then you can shorten this:

```ruby
language_skills_attributes: language_skills_attributes,
relatives_attributes: relatives_attributes,
schedules_attributes: schedules_attributes,
contact_attributes: contact_attributes
```
to:
```ruby
relatives_attributes, language_skills_attributes, schedules_attributes, contact_attributes
```

instead of the:

File: app/models/contact.rb [hunk 2]
```diff
@@ -1,15 +1,19 @@
 class Contact < ApplicationRecord
   belongs_to :contactable, polymorphic: true, optional: true
 
-  validates :name, presence: true, if: :department?
+  validates :last_name, presence: true, if: :department?
```
Comment (casaper) [inline app/models/contact.rb]: departments don't have last names.
Just keep the name field. And add first_name and last_name for the human beeings

File: app/views/contacts/_fields.slim [hunk 3]
```diff
@@ -1,5 +1,8 @@
 .row
-  .col-xs-12.col-md-6= f.input :name
+
+  - unless defined? @department
+    .col-xs-12.col-md-6= f.input :first_name
+  .col-xs-12.col-md-6= f.input :last_name, label: !defined?(@department) ? t('.last_name') : t('.name')
```
Comment (casaper) [inline app/views/contacts/_fields.slim]: just keep the name field for the departments, will make life so much easier everywhere.

File: app/views/profiles/show.html.slim [hunk 4]
```diff
@@ -1,25 +1,19 @@
 .row
   .col-xs-12
```
Comment (casaper) [inline app/views/profiles/show.html.slim]: The
```slim
.row
  .col-xs-12
```
Is bloat. It can be deleted because it has no use.

File: app/views/users/_user.html.slim [hunk 5]
```diff
@@ -1,5 +1,5 @@
 tr
-  td= link_to user.email, user
+  td=link_to user.email, user
```
Comment (casaper) [inline app/views/users/_user.html.slim]: why remove the whitespace after `=` ?

File: app/views/volunteers/_form.html.slim [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
   .col-xs-12
     = f.error_notification
-
 .row
-  .col-xs-12.col-md-6
-    = f.input :first_name
-
-  .col-xs-12.col-md-6
-    = f.input :last_name
+  .col-xs-12
```
Comment (casaper) [inline app/views/volunteers/_form.html.slim]: no need to nest the contact fields in `.row .col-xs-12` as they have their own bootstrap grid structure allready.

File: config/locales/de.yml [hunk 7]
```diff
@@ -183,6 +183,10 @@ de:
   client_updated: KlientIn wurde erfolgreich aktualisiert.
   clients: Klienten
   contact_info: Kontakt Informationen
+  contacts:
+    fields:
+      last_name: Nachname
+      name: Name
```
Comment (casaper) [inline config/locales/de.yml]: with `t_attr(:last_name)` `t_attr(:name)` in the views this whole `contacts: ...` is redundant.

File: test/system/departments_test.rb [hunk 8]
```diff
@@ -93,7 +93,7 @@ def setup
     end
     click_button 'Update Department'
     assert page.has_text? 'Department was successfully updated.'
-    assert page.has_text? '88888 88 88 88'
+    assert page.has_text? '88888888888'
```
Comment (casaper) [inline test/system/departments_test.rb]: Keep phone number like look, to catch wrong validations

File: app/views/profiles/show.html.slim (hunk 4 above)
Comment (nuritnt) [inline app/views/profiles/show.html.slim]: all views related stuff are cleaned up in this: #73

File: app/views/users/_user.html.slim (hunk 5 above)
Comment (nuritnt) [inline app/views/users/_user.html.slim]: oh, was just in my workflow, didn't even noticed it.. has no purpose.

File: test/system/departments_test.rb (hunk 8 above)
Comment (nuritnt) [inline test/system/departments_test.rb]: tests were failing due to this. was just the quick fix..

File: app/views/contacts/_fields.slim (hunk 3 above)
Comment (nuritnt) [inline app/views/contacts/_fields.slim]: with this we have the possibility to use the same columns from the contact fields and have it unified for all models

File: app/views/clients/show.html.slim [hunk 9]
```diff
@@ -1,43 +1,25 @@
 .row
   .col-xs-12
-    h1= @client.full_name
+    h1= @client.contact.full_name
```
Comment (casaper) [inline app/views/clients/show.html.slim]: same as above

File: app/views/departments/show.html.slim [hunk 10]
```diff
@@ -1,34 +1,13 @@
 .row
   .col-xs-12
-    h1= @department.contact.name
+    h1= @department.contact.last_name
```
Comment (casaper) [inline app/views/departments/show.html.slim]: unnest h1 from the grid

File: app/views/profiles/edit.html.slim [hunk 11]
```diff
@@ -2,4 +2,4 @@
   .col-xs-12
     h1= t_title(:edit)
```
Comment (casaper) [inline app/views/profiles/edit.html.slim]: unnest h1

File: app/views/profiles/show.html.slim (hunk 4 above)
Comment (casaper) [inline app/views/profiles/show.html.slim]: but this is #73

File: app/views/profiles/show.html.slim [hunk 12]
```diff
@@ -1,25 +1,17 @@
...
-    h1= @profile.full_name
+    h1= @profile.contact.full_name
 
 .row
   .col-xs-12
     .table-responsive
```
Comment (casaper) [inline app/views/profiles/show.html.slim]: unnest .table-responsive from `.row .col-xs-12`

File: app/views/volunteer_applications/thanks.html.slim [hunk 13]
```diff
@@ -2,5 +2,5 @@
   .col-xs-12.centered.columns.text-center
     h1= t('thank_you')
```
Comment (casaper) [inline app/views/volunteer_applications/thanks.html.slim]: unnest h1 to:

```slim
h1.text-center= t('thank_you')
p.text-center= t('application_submitted', email: Volunteer.last.email)
p.text-center= t('get_back')
```

File: app/views/volunteers/show.html.slim [hunk 14]
```diff
@@ -1,12 +1,13 @@
 .row
   .col-xs-12
-    h1= @volunteer.full_name
+    h1= @volunteer.contact.full_name
 
 .row
   .col-xs-12
```
Comment (casaper) [inline app/views/volunteers/show.html.slim]: unnest the table

File: config/locales/en.yml [hunk 15]
```diff
@@ -64,6 +64,10 @@ en:
   client_updated: Client was successfully updated.
   clients: Clients
   contact_info: Contact information
+  contacts:
+    fields:
```
Comment (casaper) [inline config/locales/en.yml]: t_attr use makes `contacts.fields.xxx` redundant

File: app/views/clients/show.html.slim [hunk 16]
```diff
@@ -1,43 +1,25 @@
...
     .table-responsive
       table.table.table-no-border-top
         tbody
+          = render 'contacts/show', entity: @client
           tr
             td= t_attr(:date_of_birth)
             td= l(@client.date_of_birth) if @client.date_of_birth
           tr
+            td= t_attr(:gender)
+            td= t("gender.#{@client.gender}") unless @client.gender.blank?
```
Comment (casaper) [inline app/views/clients/show.html.slim]: Not sure, but maybe?

File: app/views/volunteers/show.html.slim [hunk 17]
```diff
@@ -1,4 +1,73 @@
...
+            td= t_attr(:interests)
+            td= @volunteer.interests
+          - if policy(Volunteer).destroy?
+            tr
+              td= t_attr(:state)
+              td= t("state.#{@volunteer.state}")
+            - if @volunteer.rejected?
+              tr
+                td= t_attr(:rejection_type)
+                td= t("rejection_type.#{@volunteer.rejection_type}") unless @volunteer.rejection_type.blank?
```
Comment (casaper) [inline app/views/volunteers/show.html.slim]: if @volunteer.rejection_type.present?

File: app/views/contacts/_fields.slim [hunk 18]
```diff
@@ -1,5 +1,7 @@
 .row
-  .col-xs-12.col-md-6= f.input :name
+  - unless defined? @department
```
Comment (nuritnt) [inline app/views/contacts/_fields.slim]: why is this needed? 
or what is not good with the actual way?

File: app/views/clients/show.html.slim (hunk 16 above)
Comment (nuritnt) [inline app/views/clients/show.html.slim]: done

File: app/views/contacts/_show.html.slim [hunk 19]
```diff
@@ -0,0 +1,24 @@
+tr
```
Comment (nuritnt) [inline app/views/contacts/_show.html.slim]: hä? don't get it.

File: app/views/departments/show.html.slim [hunk 20]
```diff
@@ -1,34 +1,13 @@
...
-            td= mail_to email.body
-
-        - @department.contact.contact_phones.each do |phone|
-          tr
-            td
-              = t_attr(:body, ContactPhone)
-              = " / #{phone.label}" if phone.label.present?
-            td= phone.body
+        tbody
+          = render 'contacts/show', entity: @department
```
Comment (nuritnt) [inline app/views/departments/show.html.slim]: well than this would affect all other models and this would be refactoring, no?

File: test/system/departments_test.rb (hunk 8 above)
Comment (nuritnt) [inline test/system/departments_test.rb]: done

File: app/controllers/volunteer_applications_controller.rb [hunk 21]
```diff
@@ -24,7 +26,8 @@ def thanks; end
 
   def volunteer_params
     params.require(:volunteer).permit(
-      volunteer_attributes, language_skills_attributes, relatives_attributes, schedules_attributes
-    )
+      volunteer_attributes, language_skills_attributes, relatives_attributes, schedules_attributes,
+      contact_attributes: contact_attributes
```
Comment (nuritnt) [inline app/controllers/volunteer_applications_controller.rb]: done

File: app/views/contacts/_show.html.slim (hunk 19 above)
Comment (casaper) [inline app/views/contacts/_show.html.slim]: If you pass `, contact: @department.contact` to the partial
the partial will only need to reference `contact.first_name` instead of `entity.contact.first_name`

Comment (nuritnt) [inline app/views/contacts/_show.html.slim]: done

File: app/views/contacts/_fields.slim (hunk 18 above)
Comment (casaper) [inline app/views/contacts/_fields.slim]: Just more obvious and readable

**Comment by nuritnt** [comment]
ready for the final review @casaper @kitsane 
changes regarding unnesting/divs/form arrangement wasn't done, as I will get some feedback regarding ux & forms and can be adapted then.

**Comment by casaper** [comment]
seems good
